engine/runtime/src/main/java/org/enso/interpreter/service/ExecutionService.java

```diff
@@ -310,17 +311,25 @@ public ChangesetBuilder<Rope> modifyModuleSources(File path, List<model.TextEdit
             module.getIr(),
             TextEditor.ropeTextEditor(),
             IndexedSource.RopeIndexedSource());
+
+    Changeset<Rope> result = changesetBuilder.build(edits);
+
     JavaEditorAdapter.applyEdits(module.getLiteralSource(), edits)
         .fold(
             failure -> {
               throw new FailedToApplyEditsException(
                   path, edits, failure, module.getLiteralSource());
             },
             rope -> {
-              module.setLiteralSource(rope);
+              if (result.simpleChange() == null) {
```

Comment:
that looks pretty werid? `setLiteralSource` should just accept a pair?

Reply:
SimpleUpdate case class introduced in [a84ae04](https://github.com/enso-org/enso/pull/3508/commits/a84ae0425070de185501ea02fc54cbc9e0ab7188)